edx_oauth_client/pipeline.py

```diff
@@ -38,13 +38,18 @@ def ensure_user_information(
 
         # Received fields could be pretty different from the expected, mandatory are only 'username' and 'email'
         data['username'] = user_data.get('username', user_data.get('name'))
-        data['first_name'] = user_data.get('firstName', user_data.get('first_name'))
-        data['last_name'] = user_data.get('lastName', user_data.get('last_name'))
+        data['first_name'] = user_data.get('firstname', user_data.get('first_name'))
+        data['second_name'] = user_data.get('secondname', '')
+        data['last_name'] = user_data.get('lastname', user_data.get('last_name'))
         data['email'] = user_data.get('email')
         data['country'] = country
         data['access_token'] = access_token
+
+        if user_data.get('gender'):
+            data['gender'] = user_data.get('gender')[0]
+
         if any((data['first_name'], data['last_name'])):
-            data['name'] = u'{} {}'.format(data['first_name'], data['last_name']).strip()
+            data['name'] = '{} {} {}'.format(['first_name'], data['second_name'], data['last_name']).strip()
```

Comment:
Is it because Ironwood has `python 3` on a board?

Reply:
To the case `'{} {} {}'` I prefer to do at least this `'{0} {1} {1}'` It is more explicite as python's dzen.